src/sonic-config-engine/sonic-acl-extension.yang

```diff
@@ -7,24 +7,24 @@ module sonic-acl-extension {
   import openconfig-acl { prefix oc-acl; }
 
   typedef vlan-id-type {
-    type uint16 {
```

Comment:
>uint16 [](start = 9, length = 6)

@praveen-li, we need a nullable integer, ie, missing this field and 0 are different.
Do you know the better way?
Treat it as `string` seems not fit.

Reply:
@qiluo-msft
We can use YANG unions, Below allows only null as string input, else integer as input from 0-4095.

```
typedef vlan-id-type {
    type uint16 {
        range 0..4095;
     }
     type string {
         pattern "null";
     }
}
```

Reply:
@qiluo-msft edited above

Reply:
That seems to work, thank you @praveen-li! I will update the PR.

Reply:
@daall  Cool, if needed we can add:

default "null";

as well, so That if the user doesn't provide it, it will remain null.

Reply:
For some reason it doesn't seem like `pyangbind` handles that as expected, but if I put the "type string" clause first then it gives us the expected behavior.